src/graph.rs

```diff
@@ -460,6 +465,157 @@ where
 
         self.index = Some(index);
     }
+
+    // creates an AGraph from a list of 'good' addresses and
+    // the Edges array.
+    // For each edge, if both nodes are in the good list,
+    // a connection is created for each node.
+    // The AGraph has a equivalent information as the edges,
+    // but is just a node-centric view (required for closeness)
+    pub fn create_agraph(&self, addresses: &Vec<T>) -> AGraph {
+        let num_nodes = addresses.len();
+        let mut agraph: AGraph = AGraph::new();
+        for _ in 0..num_nodes {
+            agraph.push(Vertex::new());
+        }
+
+        // For all our edges, check if the nodes are in our address list
+        // We use the value of the addresses to find the index
+        // From then on, it's all integer indices for us
+        for edge in self.edges.iter() {
+            let source = *edge.source();
+            let target = *edge.target();
+
+            let src_result = addresses.iter().position(|&r| r == source);
+            if src_result == None {
+                continue;
+            }
+
+            let tgt_result = addresses.iter().position(|&r| r == target);
+            if tgt_result == None {
+                continue;
+            }
+
+            let src_index = src_result.unwrap();
+            let tgt_index = tgt_result.unwrap();
```

Comment:
Suggestion: maybe naming could be more consistent by using `source` and `target` here?

Reply:
done